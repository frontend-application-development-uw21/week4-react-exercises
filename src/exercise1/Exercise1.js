import React from 'react';
import CoffeeCard from './CoffeeCard';

export default class Exercise1 extends React.Component {
    render() {
        return (
            <div>
                <h2>Exercise 1</h2>
                <p>
                    In <code>Exercise1.js</code>, there is a component called <code>CoffeeCard.js</code> that is imported, but not used.
                </p>
                <ol>
                    <li>Render this component four times in <code>Exercise1.js</code> (one for each coffee drink).</li>
                    <li>Pass in the correct props.</li>
                </ol>
                <CoffeeCard drinkName="americano" price={4} />
                <CoffeeCard drinkName="latte" price={4} />
                <CoffeeCard drinkName="espresso" price={4} />
                <CoffeeCard drinkName="mocha" price={4} />
            </div>
        )
    }
}