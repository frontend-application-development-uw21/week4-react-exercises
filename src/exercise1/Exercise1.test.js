import React from 'react';
import {shallow} from 'enzyme';
import Exercise1 from './Exercise1';
import CoffeeCard from './CoffeeCard';

describe('Exercise1', () => {
    const wrapper = shallow(<Exercise1 />);
    const coffeeCards = wrapper.find('CoffeeCard');

    it('should render four CoffeeCard components', () => {
        expect(coffeeCards).toHaveLength(4);
    });

    it('should have a CoffeeCard with each drink name', () => {
        const drinkNames = coffeeCards
            .map(card => card.prop('drinkName'))
            .sort();

        expect(drinkNames).toEqual(['americano', 'espresso', 'latte', 'mocha'])
    });
});

describe('CoffeeCard', () => {
    it('should render', () => {
        const component = shallow(
            <CoffeeCard drinkName="mocha" price={4} />
        );
        expect(component).toMatchSnapshot();
    });
});