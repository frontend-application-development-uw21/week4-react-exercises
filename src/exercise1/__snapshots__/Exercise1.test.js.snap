// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoffeeCard should render 1`] = `
<div
  className="CoffeeCard"
>
  <img
    alt="mocha"
    className="CoffeeCard__image"
    src="https://upload.wikimedia.org/wikipedia/commons/7/79/Mocha.jpg"
  />
  <h3
    className="CoffeeCard__title"
  >
    Mocha
  </h3>
  <p
    className="CoffeeCard__description"
  >
    Cost: $
    4.00
  </p>
</div>
`;
